File: Testing/Stdlib Test Code/test_string_methods_out.c
#include <stdio.h>
#include <stdint.h>
#include <string.h>

int str_len(char* s);
int str_contains(char* s, char* sub);
int main(void);

int str_len(char* s)
{
    return strlen(s);
}

int str_contains(char* s, char* sub)
{
    char* found = strstr(s, sub);
    return (found != NULL);
}

int main(void)
{
    char* s = "Hello World";
    int len = str_len(s);
    printf("%d\n", len);
    int has = str_contains(s, "World");
    printf("%d\n", has);
    return 0;
}

File: Testing/Stdlib Test Code/test_string_methods_out.js
function strlen(s) { return s.length; }
function strstr(s, sub) { const i = s.indexOf(sub); return i >= 0 ? s.slice(i) : null; }

function str_len(s) {
    return strlen(s);
}

function str_contains(s, sub) {
    let found = strstr(s, sub);
    return (found !== null);
}

function main() {
    let s = "Hello World";
    let len = str_len(s);
    console.log(len);
    let has = str_contains(s, "World");
    console.log(has);
    return 0;
}


main();
